### PR TITLE
rework database creation, cleanup + other db fixes

### PR DESCRIPTION
**Pull request by rutsky**

### CODE REVIEW

**Comment by rutsky** [comment]
I merged Travis config from PR #1982 to this PR.
PR #1982 shows, that there are errors when file-based SQLite database is used, and after merging of PR #1982 commits we should see, that there is no more errors (or at least less number of errors :) ) --- so this PR is actually fixing something.

File: .travis.yml [hunk 1]
```diff
@@ -14,6 +14,7 @@ env:
   - TWISTED=13.0.0 SQLALCHEMY=latest TESTS=trial
   - TWISTED=15.4.0 SQLALCHEMY=latest TESTS=trial
   - TWISTED=latest SQLALCHEMY=latest TESTS=trial
+  - TWISTED=latest SQLALCHEMY=latest TESTS=trial BUILDBOT_TEST_DB_URL=sqlite:////tmp/test_db.sqlite
```
Comment (djmitche) [inline .travis.yml]: Can you include a comment about why this is useful, compared to the default memory DB?

File: master/buildbot/test/util/db.py [hunk 2]
```diff
@@ -65,16 +65,76 @@ class RealDatabaseMixin(object):
...
+                    buildsets_table = meta.tables['buildsets']
+                    for fkc in buildsets_table.foreign_key_constraints:
+                        if fkc.referred_table.name == 'builds':
+                            fkc.use_alter = True
+
+                if 'builds' in meta.tables:
+                    builds_table = meta.tables['builds']
+                    for fkc in builds_table.foreign_key_constraints:
+                        if fkc.referred_table.name == 'buildrequests':
+                            fkc.use_alter = True
```
Comment (djmitche) [inline master/buildbot/test/util/db.py]: I'd rather have this read from some data structure like `[('buildsets', 'builds'), ('builds', 'buildrequests')]`

**Comment by djmitche** [comment]
This looks good aside from the stylistic comment.

**Comment by rutsky** [comment]
@djmitche, I fixed issues that you noted.

**Comment by rutsky** [comment]
@djmitche, is my fixes looks good to you?

**Comment by djmitche** [comment]
@rutsky perfect, yes

**Pull request merged**